gulpfile.js/import/importRoadmap.js

```diff
@@ -98,33 +127,50 @@ async function importRoadmap() {
       }
 
       roadmap.push({
-        'wg_name': workingGroupName,
+        'wg_slug': workingGroupSlug,
+        'wg_name': meta.title,
         'created_at': createdAt,
-        'status_update': statusUpdate || '',
-        'quarter': quarter || '',
-        'title': title,
+        'status_update': statusUpdate,
+        'quarter': quarter,
         'number': issue.number,
-        'author': issue.user.login,
         'html_url': issue.html_url,
-        'body': issue.body,
+        'body': body,
       });
     }
 
     log.info(`.. ${wg.name} - ${issues.length} issues imported`);
   }
 
-  // Sort issues by date
+  // Get full name from working group meta.yaml and predefined color based on roadmap config
+  workingGroups = [...new Set(workingGroups.sort())];
+  workingGroups = workingGroups.map((group, index) => {
+    return {
+      'slug': group.slug,
+      'name': group.name,
+      'color': config.workingGroupColors[index] || '#000',
+    };
+  });
+
+  // Sort roadmap entries and add matching working group color
   roadmap = roadmap.sort((a, b) => {
     return new Date(b.status_update) - new Date(a.status_update);
   });
+  for (const group of workingGroups) {
```

Comment:
This nested loop is ugly. You're looping over all the issues for every working group.

When you restructure the config object as suggested above you could simply loop over the issues once, get rid of the `if` and do something along the lines of:

```js
for (const issue of roadmap) {
 issue.color = config.colors[issue.wg_slug];
}
```